dhcpv4/option_subnet_mask.go

```diff
@@ -0,0 +1,56 @@
+package dhcpv4
+
+import (
+	"fmt"
+	"net"
+)
+
+// This option implements the subnet mask option
+// https://tools.ietf.org/html/rfc2132
+
+// OptSubnetMask represents an option encapsulating the subnet mask.
+type OptSubnetMask struct {
+	SubnetMask net.IPMask
+}
+
+// ParseOptSubnetMask returns a new OptSubnetMask from a byte
+// stream, or error if any.
+func ParseOptSubnetMask(data []byte) (*OptSubnetMask, error) {
+	if len(data) < 2 {
```

Comment:
nit: the length should be always 6 here. This would remove the need for the redundant check below that `len(data) == 6`. From the RFC:

> The code for the subnet mask option is 1, and its length is 4 octets.